### PR TITLE
refactor(material/sort): explicit sort direction

### PR DESCRIPTION
**Pull request by lekhmanrus**
Not sure if that is really needed.
Just it feels like it is more consistent with existing `SortDirection`. 🙂

### CODE REVIEW

File: src/material/sort/sort-header.ts [hunk 1]
```diff
@@ -130,7 +130,7 @@ export class MatSortHeader extends _MatSortHeaderBase
   @Input() arrowPosition: 'before' | 'after' = 'after';
 
   /** Overrides the sort start value of the containing MatSort for this MatSortable. */
-  @Input() start: 'asc' | 'desc';
+  @Input() start: SortDirectionExplicit;
```
Comment (crisbeto) [inline src/material/sort/sort-header.ts:133]: The current code in master seems incorrect to me. I would've expected it to be a `SortDirection`. @andrewseguin do you have more context on this?

**Comment by lekhmanrus** [comment]
Is there a way to restart [ci/circleci: tests_browserstack](https://app.circleci.com/pipelines/github/angular/components/29907/workflows/fa76b90b-8ee1-40ad-904b-31c5561ca1fc/jobs/313972) test?

**Comment by lekhmanrus** [comment]
Looks like the failing test is unrelated to the change. Could we rerun it again, please?

**Review by andrewseguin — APPROVED** [review]

**Comment by andrewseguin** [comment]
I can't remember if we wanted these to explicitly NOT be the empty string, but I can't think of a good enough justification now for it

**Pull request merged**